### DISCUSSION TITLE
Scrolling issue with NativeControl in Avalonia 11

### DISCUSSION THREAD

**Discussion by mohdimas** [AvaloniaUI/Avalonia#15024] [Q&A]
Hi all,

Since upgrading to Avalonia 11.x we encountered issue when scrolling. See the [screencast](https://www.screencast.com/t/gwKIavTFVjsD). There's a delay or speed difference between the Avalonia control and the native control when scrolling. So it results in a "bouncy" effect when scrolling up and down.

![Kapture 2024-03-18 at 19 13 34](https://github.com/AvaloniaUI/Avalonia/assets/1266072/ea6d3340-e475-480f-899c-33277e782328)

The treeview is like this
```

<ScrollViewer>
      <ItemsControl>
            <ItemTemplate>
                  <StackPanel>
                        <ContentControl>..Header content..</ContentControl>
                        <CEFBrowser />
                  </StackPanel>
            </ItemTemplate>
      </ItemsControl>
</ScrollViewer>
```

<img width="777" alt="Screenshot 2024-03-18 at 6 59 11 PM" src="https://github.com/AvaloniaUI/Avalonia/assets/1266072/8f9e97bb-8e6a-4618-a5f3-b226a73bf939">

It's tricky to create a sample project, but maybe you have ideas why it may happen?

Many thanks!

**Comment by timunie** [comment]
ItemsControl is virtualized by default now. Maybe try to set ItemPanel to StackPanel again to disable virtualization in your case

**Reply by mohdimas** [reply]
hi,
thanks for the reply. I tried that before but unfortunately it doesnt work. I also tried using different rendering mode (`OpenGL` and `Software`, but didnt work. I also played around with the `IsScrollInertiaEnabled` property but it was the same, it didn't work. 

The only thing that works is to use `OffScreen` for the CEF, but unfortunately it wasn't possible to use `OffScreen`. Do you may have other idea?

**Reply by mohdimas** [reply]
One of our developers found out that the issue may be related to this line of [code](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/NativeControlHost.cs#L146). If it's changed back to `Render`, with some additional tweak then the issue is less noticeable.

**Comment by mohdimas** [comment]
We added 2 test projects [here](https://github.com/mohdimas/AvaloniaNativeControlEmbedIssue). The first one is how it behaves with the `Avalonia 11` and the other one is with `Avalonia 0.10`. In our test the `treeview` is like this
```
<ScrollViewer>
        <StackPanel>
            <Red Panel />
            <Blue Panel />
            <Black NativeControl />
            <Blue Panel />
            <Red Panel />
        </StackPanel>
</ScrollViewer>
```

In `Avalonia 11` the black rectangle, which is a `NativeControl`, is overlapping with the items after and moving at different speed than the other rectangles. Notice that there's a noticeable white gap in between the black and the blue rectangle. When scrolling up and down it caused a `bouncy` effect which doesn't look right.

Meanwhile in `Avalonia 0.10`, the black rectangle doesn't overlap or move far away from the blue rectangle. When there's a gap between the black and blue rectangle, the black rectangle quickly moved to the correct position to close the gap. See gif below:

**Version 11**
![Kapture 2024-03-21 at 16 06 47](https://github.com/AvaloniaUI/Avalonia/assets/1266072/d8da4a24-e215-49ac-bbb6-4079b50b2cb7)


**Version 0.10**
![Kapture 2024-03-21 at 16 28 11](https://github.com/AvaloniaUI/Avalonia/assets/1266072/956d00c6-c6b4-4306-b0b1-08843c873f96)

@timunie do you have any thought on this? Maybe you can point us to which part of the code that was changed in the update.
Many thanks 🙏
